### ISSUE TITLE
Triplet loss produces nans

### ISSUE THREAD

**Issue by dkajtoch** [KevinMusgrave/pytorch-metric-learning#107]
I am using TripletMarginLoss and there are situations during training when the total number of triplets is exactly zero. If option `avg_non_zero_only` is used then loss becomes `nan` in my setup. A quick fix below projects loss to zero as it should be:
```
self.num_non_zero_triplets = (relued > 0).nonzero().size(0)
   if self.avg_non_zero_only and self.num_non_zero_triplets > 0:
      return torch.sum(relued) / self.num_non_zero_triplets
   return torch.mean(relued)
```

**Comment by KevinMusgrave** [comment]
Thanks, I made the change you suggested in TripletMarginLoss, ContrastiveLoss, and SignalToNoiseRatioContrastiveLoss. [Here's the commit](https://github.com/KevinMusgrave/pytorch-metric-learning/commit/00b65b62716b6cb87e677f24a1bbc5f8ae4d64fb).

You can get the changes by installing 0.9.87.dev1:
```
pip install pytorch-metric-learning==0.9.87.dev1
```
